### DISCUSSION TITLE
GO  with Visual Studio 2019

### DISCUSSION THREAD

**Discussion by larspalo** [GrandOrgue/grandorgue#365] [General]
Originally posted 2020-12-31 15:43:35.347000 by vpo in GrandOrgue discussion forum on SourceForge<br><br>Hello all,
Is there a Visual Studio 2019 solution for the GO project?
Greetings
vpo

**Comment by larspalo** [comment]
Originally posted 2021-01-04 19:54:54.654000 by dvdgrttr in GrandOrgue discussion forum on SourceForge<br><br>are you trying to compile for windows or linux?  If for linux, I suggest you try visual studio code as an editor/simple ide.  It runs natively on linux and   there is documentation for using Cmake/ Make under visual studio code, or there is the possibility of using just the editor with its ability to austocomplete/ cross reference functions etc, and continue to use cmake and make from the command line to compile the project. You will note the recommended way to compile for windows is to cross compile under linux  as well

**Reply by larspalo** [reply]
Originally posted 2021-01-05 07:37:42.201000 by vpo in GrandOrgue discussion forum on SourceForge<br><br>I&#039;m compile for Windows under Win10.
Are there any instructions for the Linux cross-compile? Maybe I can derive from it how to build a VS2019 solution.

**Comment by larspalo** [comment]
Originally posted 2021-01-05 15:07:33.351000 by [csw900](https://github.com/midimusic) in GrandOrgue discussion forum on SourceForge<br><br>Why do you need to compile GO for Windows? There is a perfectly good compiled version already available for download. That is what I use. I think this Windows version was compiled on Linux and it will work on any 64 bit windows system.

If I was compiling GO on Windows I would use cmake and mingw. (However the GO source code lacks CMakeLists.txt which Windows cmake needs but  Linux cmake does not). Trying to use Visual Studio makes unnecessary difficulties.

However in the unlikely event you get GO to compile on Visual Studio please tell us how to do it.

csw900

**Comment by larspalo** [comment]
Originally posted 2021-01-05 15:49:29.405000 by vpo in GrandOrgue discussion forum on SourceForge<br><br>I am a C++ developer with over 20 years of experience and would like to extend GO or support its development.

&gt; However in the unlikely event you get GO to compile on Visual Studio please tell us how to do it

Is this a joke? :-)
I thought you were the GO project specialists. I tried to compile GO a few years ago and gave up at some point.

**Comment by larspalo** [comment]
Originally posted 2021-01-07 16:54:47.685000 by dvdgrttr in GrandOrgue discussion forum on SourceForge<br><br>to answer the question  posed above, the instructions for compiling on windows and cross compiling for windows using linux are both found in  the README file found in the root directory of the source tree.

**Comment by larspalo** [comment]
Originally posted 2021-01-08 15:11:36.296000 by [csw900](https://github.com/midimusic) in GrandOrgue discussion forum on SourceForge<br><br>Thanks david gritter.
Very useful information in this readme - note there is no mention of VisualStudio. There is some very good information on setting up a windows system to compile normal cMake based code. You will need to compile a 64 bit version of GO in order to use the larger samplesets.
csw900

**Comment by larspalo** [comment]
Originally posted 2021-01-09 15:37:57.208000 by sandstone42 in GrandOrgue discussion forum on SourceForge<br><br>The only reason I learned to &quot;build my own&quot; was to make some minor image changes (using my own drawknobs and backgrounds.) If you&#039;re looking for a Visual Studio solution, you&#039;ll probably have to develop your own. I think that at one time there was a build using the Code Blocks IDE but it was never supported long term.

**Reply by larspalo** [reply]
Originally posted 2021-01-10 11:07:19.841000 by [Lars Palo](https://github.com/larspalo) in GrandOrgue discussion forum on SourceForge<br><br>&gt; The only reason I learned to &quot;build my own&quot; was to make some minor image
&gt; changes (using my own drawknobs and backgrounds.)
&gt;
Honestly, if you only wanted to add your own graphics there&#039;s no need to
build GO yourself... That possibility has been in GO for a very long time!
You just need to learn a little about the ODF format.

&gt; If you&#039;re looking for a Visual Studio solution, you&#039;ll probably have to
&gt; develop your own. I think that at one time there was a build using the Code
&gt; Blocks IDE but it was never supported long
&gt;
At a certain point, long time ago, I did build GO with CodeBlocks but that
was not really so useful for me so I abandoned it and instead switched to
using just a simple text editor and some command line tools.

However, I wouldn&#039;t expect it to be impossible to use an IDE for
development work. I just don&#039;t need it. Best advice is likely that if you
want it -you fix it yourself.

&gt;
<br><br>Attachments in original post:<br>https://sourceforge.net/p/ourorgan/discussion/962124/thread/bc2d508eb3/f7fc/5aec/attachment/alternate<br>

**Reply by larspalo** [reply]
Originally posted 2021-01-10 19:47:57.652000 by vpo in GrandOrgue discussion forum on SourceForge<br><br>If you don&#039;t know the good, you can&#039;t miss it...

**Comment by larspalo** [comment]
Originally posted 2021-01-10 20:53:29.600000 by vpo in GrandOrgue discussion forum on SourceForge<br><br>I&#039;ve compiled wxWidgets, CMake and TDM installed, and CMake&#039;s first error is:
Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE)
Do I have to install this, or is there just a missing path (on Windows)?

**Comment by larspalo** [comment]
Originally posted 2021-01-10 21:10:30.501000 by jlder in GrandOrgue discussion forum on SourceForge<br><br>If you look under svn, you will find all the history and the fact GrandOrgue was initially using CodeBlocks.
While resurrecting myorgan and transitioning to GrandOrgue, it took some efforts to get CodeBlocks running.
I personally enjoyed working on GrandOrgue using an IDE.
Martin, who did a lot of work and an outstanding job on GrandOrgue, did not need an IDE.
The bottom line is that if you want/need to develop GrandOrgue, it is up to you to decide if it is a good trade to invest the time in porting  GrandOrgue to a specific IDE, in order to save  time  while developing the code.

**Reply by larspalo** [reply]
Originally posted 2021-01-10 22:25:31.209000 by vpo in GrandOrgue discussion forum on SourceForge<br><br>I don&#039;t want programming like in the Stone Age. I did that 30 years ago. I have now installed CodeBlocks 20.3. How do I get GO to run with it?
I can&#039;t get the project to work with CMake on Windows. Is it easier now with CodeBlocks? Is there a project file?

**Comment by larspalo** [comment]
Originally posted 2021-01-10 23:07:53.635000 by jlder in GrandOrgue discussion forum on SourceForge<br><br>GrandOrgue CodeBlocks time was 10 years ago and I did not keep the configuration files (Maybe other admin might have kept them)
In any case they would probably require a lot of cleanup.
I guess you will need to do the homework if you want to use an IDE.
You can probably use VS to edit the code and compile GO under Windows using WSL (see the TUTO)

**Comment by larspalo** [comment]
Originally posted 2021-01-11 18:48:29.860000 by [Oleg Samarin](https://github.com/oleg68) in GrandOrgue discussion forum on SourceForge<br><br>I use NetBeans - a freeware IDE https://netbeans.org/  for making changes, compiling and debugging GrandOrgue. Creating a GrandOrgue C++ project with existing source is quite simple with NetBeans. Everything works excellent exept I run Linux on my desktop compuer and I  compile GrandOrgue for Linux.

Netbeans itself is a cross-platform IDE (written in Java) and can be run on Windows. But I&#039;ve not tested if it can compile GrandOrgue for Windows.
